### PR TITLE
Changed nesy pattern syntax

### LINKED ISSUE #2075: Change of NeSy pattern syntax
Currently, we can write `class`, `class[id]` or `id`. Please switch to `class`, `id : class`, because this conforms better with OWL2 ABox syntax. (The third option with `id` should go.)

### PR DESCRIPTION
**Pull request by b-gehrke**
Closes #2075

### CODE REVIEW

File: NeSyPatterns/AS.der.hs [hunk 1]
```diff
@@ -32,10 +32,10 @@ import Data.Data
 -- DrIFT command
 {-! global: GetRange !-}
 
--- | nodes are of form: ontology_term[id]
--- | both components are optional, but at least one must be present
+-- | nodes are of form: id : ontology_term
+-- | id is optional
 data Node = Node {
-    ontologyTerm :: (Maybe Id.Token),
+    ontologyTerm :: Id.Token,
```
Comment (mcodescu) [inline NeSyPatterns/AS.der.hs:38]: Why not IRI instead of Token?

Comment (b-gehrke) [reply NeSyPatterns/AS.der.hs:38]: I remember @tillmo mentioning this should be done in a later task. Is this still true or can it done in this?

Comment (mcodescu) [reply NeSyPatterns/AS.der.hs:38]: I have a workaround in the colimit computation, so I guess it could stay. @tillmo what do you think?

Comment (tillmo) [reply NeSyPatterns/AS.der.hs:38]: If you have a workaround for colimits, let us stay with Token here, and have a separate issue for changing it into IRI.

**Review by mcodescu — APPROVED** [review]
Looks good to me.

**Review by tillmo — CHANGES_REQUESTED** [review]
looks good, but I would like to be able to write `m :  Model`. Currently, I need to write `m: Model`, i.e. without spaces.

**Comment by b-gehrke** [comment]
> looks good, but I would like to be able to write `m : Model`. Currently, I need to write `m:Model`, i.e. without spaces.

Fixed

**Pull request merged**